src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -265,7 +268,9 @@ impl<S: Snapshot, P: ScanPolicy<S>> ForwardScanner<S, P> {
             };
 
             if has_lock {
-                if self.met_newer_ts_data == NewerTsCheckState::NotMetYet {
+                if self.met_newer_ts_data == NewerTsCheckState::NotMetYet
+                    && self.cfg.isolation_level == IsolationLevel::Si
+                {
```

Comment:
Why not set `met_newer_ts_data` for `IsolationLevel::RcCheckTs`?

Reply:
The scanner does not rely on this field to check the latest data so it's not set.

Reply:
It doesn't matter in `RcCheckTs` case, and pointGetter also does not set it, but it's set in the backward scan 🤔.

Reply:
I'll change the backward scanner not to set it to keep.

Reply:
I think it more sematically correct to set it to `NewerTsCheckState::Met` here no matter the isolation level is...

Reply:
I've changed the logic not to depend on the `NewerTsCheckState` related behaviours and the default behaviours of it would not be affected.